xmppserver/src/main/java/org/jivesoftware/openfire/handler/IQVersionHandler.java

```diff
@@ -76,6 +83,40 @@ public IQ handleIQ(IQ packet) throws PacketException {
             IQ result = IQ.createResultIQ(packet);
             result.setError(PacketError.Condition.not_acceptable);
             return result;
+        } else if (IQ.Type.result == packet.getType()) {
+            /* handle results coming through BOSH Connections,
+             * other results are processed in org.jivesoftware.openfire.net.SocketRead.java - getIQ()
+             */
+            try {
+                LocalSession localSession = (LocalSession) XMPPServer.getInstance().getSessionManager().getSession(packet.getFrom());
+
+                Element query = packet.getChildElement();
+                List<Element> elements = query.elements();
+                if (elements.size() > 0) {
+                    for (Element element : elements) {
+                        if (element.getName() != null && element.getStringValue() != null) {
+                            if (localSession!=null)
+                            {
+                                localSession.setSoftwareVersionData(element.getName(), element.getStringValue());
+                            }
+                            else
+                            {
+                                /*
+                                  The result comes from a server 2 server connection, so we write the information
+                                  to the log, because we dont need it at this point.
+                                */
+                                Log.info(packet.getFrom()+" "+element.getName()+"="+element.getStringValue());
+                            }
+                        } else {
+                            Log.warn("No software version data found in packet: " + packet.toXML());
+                        }
+                    }
+                }
+
+            } catch (Exception e) {
```

Comment:
What are you trying to defend against here? Could this be more explicit?

Reply:
oh long time ago:D i dont know exactly why .. but i think it was for development too... so i will remove it